src/main/java/com/quantumsoft/qupathcloud/converter/dicomizer/Dicomizer.java

```diff
@@ -0,0 +1,163 @@
+package com.quantumsoft.qupathcloud.converter.dicomizer;
+
+import com.sun.jna.NativeLong;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dcm4che3.util.UIDUtils;
+
+import java.io.IOException;
+import java.nio.DoubleBuffer;
+import java.nio.charset.StandardCharsets;
+
+public class Dicomizer {
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    public void run(Options options) throws IOException {
+        int exitCode = Wsi2dcmLibrary.INSTANCE.wsi2dcm(
+                StandardCharsets.UTF_8.encode(options.getInputPath()),
+                StandardCharsets.UTF_8.encode(options.getOutputFolder()),
+                new NativeLong(options.getTileWidth()),
+                new NativeLong(options.getTileHeight()),
+                StandardCharsets.UTF_8.encode(options.getCompression().getValue()),
+                options.getCompressionQuality(),
+                0,
+                -1,
+                StandardCharsets.UTF_8.encode(options.getImageName()),
+                StandardCharsets.UTF_8.encode(UIDUtils.createUID()),
+                StandardCharsets.UTF_8.encode(UIDUtils.createUID()),
+                options.getPyramidLevels(),
+                DoubleBuffer.wrap(options.getDownsamples()),
+                (byte)1,
+                500,
+                options.getThreadCount(),
+                (byte)0);
+        if (exitCode != 0) {
+            throw new IOException("Dicomizer error, exit code: " + exitCode);
+        }
+    }
+
+    public static class Options {
+        private String inputPath;
+        private String outputFolder;
+        private String imageName;
+
+        private Integer threadCount = -1; // all available
+        private Integer pyramidLevels = 0; // use as is
+        private Integer tileWidth = 500;
+        private Integer tileHeight = 500;
+        private Compression compression = Compression.JPEG;
+        private int compressionQuality = 80;
+
+        public Options inputPath(String value) {
+            inputPath = value;
+            return this;
+        }
+
+        public Options threads(int count) {
+            // default 2
+            threadCount = count;
+            return this;
+        }
+
+        public Options generatePyramid(int levels) {
+            pyramidLevels = levels;
+            return this;
+        }
+
+        public Options tileWidth(int width) {
+            // Width of the tiles in the target image
```

Comment:
These comments look like they should be Javadoc comments instead.

Reply:
Ok, will change.